### PR TITLE
apps/speed.c: add -aead flag.

### PR DESCRIPTION
**Pull request by dot-asm**
Goal is to exercise AEAD ciphers in TLS-like sequence, i.e. 13-byte
AAD followed by payload.
    
[While we are at it, address even some styling issues.]

### CODE REVIEW

**Comment by dot-asm** [comment]
I had a brainslip, pushed update...

**Comment by dot-asm** [comment]
Red cross from travis is unrelated, and [should be] addressed in #6315.

File: apps/speed.c [hunk 1]
```diff
@@ -2430,14 +2474,25 @@ int speed_main(int argc, char **argv)
...
-                 * -O3 -fschedule-insns messes up an optimization here!
-                 * names[D_EVP] somehow becomes NULL
-                 */
+        loopfunc = EVP_Update_loop;
+        if (evp_cipher != NULL) {
+            names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
+
+            if (EVP_CIPHER_mode(evp_cipher) == EVP_CIPH_CCM_MODE) {
+                loopfunc = EVP_Update_loop_ccm;
+            } else if (aead && (EVP_CIPHER_flags(evp_cipher) &
```
Comment (FdaSilvaYY) [inline apps/speed.c]: Why not apply this to every cipher having EVP_CIPH_FLAG_AEAD_CIPHER flag ?

File: apps/speed.c [hunk 2]
```diff
@@ -327,6 +328,8 @@ const OPTIONS speed_options[] = {
      "Run benchmarks for pnum seconds"},
     {"bytes", OPT_BYTES, 'p',
      "Run cipher, digest and rand benchmarks on pnum bytes"},
+    {"aead", OPT_AEAD, '-',
```
Comment (FdaSilvaYY) [inline apps/speed.c]: this flag must be used in conjonction with -evp , isn't it ?

File: apps/speed.c [hunk 3]
```diff
@@ -2436,14 +2474,25 @@ int speed_main(int argc, char **argv)
...
-                 * -O3 -fschedule-insns messes up an optimization here!
-                 * names[D_EVP] somehow becomes NULL
-                 */
+        loopfunc = EVP_Update_loop;
+        if (evp_cipher != NULL) {
+            names[D_EVP] = OBJ_nid2ln(EVP_CIPHER_nid(evp_cipher));
+
+            if (EVP_CIPHER_mode(evp_cipher) == EVP_CIPH_CCM_MODE) {
+                loopfunc = EVP_Update_loop_ccm;
+            } else if (aead && (EVP_CIPHER_flags(evp_cipher) &
```
Comment (FdaSilvaYY) [inline apps/speed.c]: is there a reason to use an cipher having an AEAD flag in another way than AEAD ?

File: apps/speed.c (hunk 1 above)
Comment (dot-asm) [reply apps/speed.c]: Yes, I considered this. Trouble is that those who want to profile low-level subroutines would prefer the original behaviour. So there would be additional flag either way. Being conservative in minor release is more appropriate, i.e. it's more appropriate to engage new behaviour with additional flag and not vice versa.

File: apps/speed.c (hunk 2 above)
Comment (dot-asm) [reply apps/speed.c]: Not only that, but with cipher. I mean -evp can denote digest as well. Yeah, maybe sanity check with an error message would be appropriate...

File: apps/speed.c (hunk 1 above)
Comment (FdaSilvaYY) [reply apps/speed.c]: Ok, I see.

File: apps/speed.c (hunk 2 above)
Comment (FdaSilvaYY) [reply apps/speed.c]: I suggest to take inspiration on the 'mb' option, a few lines above ::
```
     {"mb", OPT_MB, '-',
      "Enable (tls1.1) multi-block mode on evp_cipher requested with -evp"},
```

Comment (dot-asm) [reply apps/speed.c]: Like that?

File: apps/speed.c [hunk 4]
```diff
@@ -948,6 +951,41 @@ static int EVP_Update_loop_ccm(void *args)
     }
     return count;
 }
+/*
```
Comment (FdaSilvaYY) [inline apps/speed.c]: style : blank line to mark separation ?

**Pull request closed without merging**